stdlib/public/core/OutputStream.swift

```diff
@@ -330,9 +330,17 @@ internal struct _Stdout : OutputStreamType {
   }
 
   mutating func write(string: String) {
-    // FIXME: buffering?
-    // It is important that we use stdio routines in order to correctly
-    // interoperate with stdio buffering.
+    if string.isEmpty { return }
+
+    if string._core.isASCII {
+      let result = _swift_stdlib_fwrite_stdout(
+        UnsafePointer(string._core.startASCII), string._core.count, 1)
```

Comment:
I was under the impression that ARC guarantees that it doesn't release any values until the end of the full statement. Maybe that only applies to Obj-C ARC and not Swift?